File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/Go/s196249385_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	pr := bufio.NewWriter(os.Stdout)
	defer pr.Flush()

	solve(sc, pr)
}

func solve(sc *bufio.Scanner, pr *bufio.Writer) {
	sc.Scan()
	n, _ := strconv.Atoi(sc.Text())
	sc.Scan()
	k, _ := strconv.Atoi(sc.Text())

	a := make([]int, n)
	for i := 0; i < n; i++ {
		sc.Scan()
		a[i], _ = strconv.Atoi(sc.Text())
	}

	cumsum := make([]int64, n+1)
	for i := 0; i < n; i++ {
		cumsum[i+1] = cumsum[i] + int64(a[i])
	}

	b := make([]int64, 0, n*(n+1)/2)
	var max int64
	for i := 0; i < n; i++ {
		for j := i; j < n; j++ {
			diff := cumsum[j+1] - cumsum[i]
			b = append(b, diff)
			if diff > max {
				max = diff
			}
		}
	}

	mask := int64(1)
	for mask <= max {
		mask <<= 1
	}
	mask >>= 1

	for mask > 0 {
		var tmp []int64
		for _, e := range b {
			if e&mask != 0 {
				tmp = append(tmp, e)
			}
		}

		if len(tmp) >= k {
			b = tmp
		}

		mask >>= 1
	}

	if len(b) < k {
		fmt.Fprintln(pr, 0)
	} else {
		ans := int64(^uint64(0) >> 1)
		for _, e := range b {
			ans &= e
		}

		fmt.Fprintln(pr, ans)
	}
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/C/s196249385_commented.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_N 1000

int main() {
    int n, k;
    scanf("%d %d", &n, &k);

    int a[MAX_N];
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }

    long cumsum[MAX_N + 1] = {0};
    for (int i = 0; i < n; i++) {
        cumsum[i + 1] = cumsum[i] + a[i];
    }

    long *b = (long *)malloc((n * (n + 1) / 2) * sizeof(long));
    int b_size = 0;
    long max = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            b[b_size++] = cumsum[j + 1] - cumsum[i];
            if (b[b_size - 1] > max) {
                max = b[b_size - 1];
            }
        }
    }

    long mask = 1;
    while (mask <= max) {
        mask <<= 1;
    }
    mask >>= 1;

    while (mask > 0) {
        int tmp_size = 0;
        long *tmp = (long *)malloc(b_size * sizeof(long));
        for (int i = 0; i < b_size; i++) {
            if (b[i] & mask) {
                tmp[tmp_size++] = b[i];
            }
        }
        free(b);
        b = tmp;
        b_size = tmp_size;
        mask >>= 1;
    }

    if (b_size < k) {
        printf("0\n");
    } else {
        long ans = -1L;
        for (int i = 0; i < b_size; i++) {
            ans &= b[i];
        }
        printf("%ld\n", ans);
    }

    free(b);
    return 0;
}
// End of Code
